src/Main.elm

```diff
@@ -1,10 +1,197 @@
+                        , breadCrumbId = newBreadcrumbId id
+                    }
+            }
+
+        MsgFolderClosed ->
+            { model
+                | opened =
+                    { newOpened
+                        | currentId = newCurrentId
+                        , parentId = List.head (List.drop 2 newOpened.usedIdList)
+                        , parentName = List.drop 1 newOpened.parentName
+                        , usedIdList = List.drop 1 newOpened.usedIdList
+                        , state = stateOfcurrentId newCurrentId
+                    }
+                , breadcrumbs =
+                    { newBreadcrumb
+                        | breadCrumbName = List.drop 1 newBreadcrumb.breadCrumbName
+                        , breadCrumbId = List.drop 1 newBreadcrumb.breadCrumbId
+                    }
+            }
+
+
+view : Model -> Html Msg
+view model =
+    let
+        isOpened =
+            model.opened.state
+    in
+    Html.div [ class "container" ]
+        [ Html.h1 [ class "h1" ]
+            [ Html.text "Audience Browser"
+            ]
+        , Html.div [ class "listContainer" ]
+            [ Html.div [ id "head" ]
+                [ Html.div [ class "breadcrumbsArea" ]
+                    [ Html.a [ class "breadcrumbsText" ] [ text "Home" ]
+                    , Html.ul [ class "breadcrumbsText" ] (List.map (breadCrumbs model.breadcrumbs) (List.reverse model.breadcrumbs.breadCrumbName))
+                    ]
+                , if isOpened == True then
+                    Html.button [ class "backButton", onClick MsgFolderClosed ] [ text "Go Up" ]
+
+                  else
+                    text ""
+                ]
+            , Html.ul [ class "list" ]
+                (List.map (openFolder model.opened) model.folders)
+            , Html.ul [ class "list" ]
+                (List.map (viewAudience model.opened) model.audience)
+            ]
+        ]
+
+
+openFolder : Opened -> AudFolders.AudienceFolder -> Html Msg
+openFolder opened list =
+    let
+        folderButton =
+            Html.button [ class "folder", onClick (MsgFolderOpened list.id list.name) ] [ text list.name ]
+    in
+    if opened.currentId == list.parent && opened.state == True then
+        folderButton
+
+    else if list.parent == Nothing && opened.state == False then
+        folderButton
+
+    else
+        text ""
+
+breadCrumbs : Breadcrumbs -> String -> Html msg
+breadCrumbs breadcrumbs list =
+    li [ class "breadcrumbs" ] [ text (" -> " ++ list) ]
+
+
+viewAudience : Opened -> Aud.Audience -> Html msg
+viewAudience opened listAudience =
+    let
+        audienceButton =
+            Html.button [ class "audience" ] [ text listAudience.name ]
+    in
+    if opened.currentId == listAudience.folder && opened.state == True then
+        audienceButton
+
+    else if listAudience.folder == Nothing && opened.state == False then
+        audienceButton
+
+    else
+        text ""
```

Comment:
If you get rid of `state` you can simply do this, because of `folder: Maybe Int`.
```suggestion
    if opened.currentId == listAudience.folder then
        audienceButton

    else
        text ""
```